app/src/main/java/org/jellyfin/mobile/player/deviceprofile/DeviceProfileBuilder.kt

```diff
@@ -13,8 +14,11 @@ import org.jellyfin.sdk.model.api.SubtitleDeliveryMethod
 import org.jellyfin.sdk.model.api.SubtitleProfile
 import org.jellyfin.sdk.model.api.TranscodeSeekInfo
 import org.jellyfin.sdk.model.api.TranscodingProfile
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
 
-class DeviceProfileBuilder {
+class DeviceProfileBuilder : KoinComponent {
+    private val appPreferences: AppPreferences by inject()
```

Comment:
You should be able to use constructor injection here. On top of my suggestion, you need to add a `get()` parameter in the AppModule.

```suggestion
class DeviceProfileBuilder(
    private val appPreferences: AppPreferences,
) {
```